From 487fe77a16b1985cb4681d7607ec25033888f06f Mon Sep 17 00:00:00 2001
From: Ricard Wanderlof <polluxsynth@butoba.net>
Date: Wed, 9 Feb 2022 23:41:19 +0100
Subject: [PATCH 1844/1936] Bank.cpp: Update database after write (#331)

When writing a patch to a bank slot, rescan the database to keep
it updated. Otherwise, a subsequent search() call will not reflect
the current state of the banks, and consequently switching to Read
mode in the UI after writing to a slot in Write mode causes the old
bank contents to be displayed, and also causes any attempt to
load the recently saved patch to fail, as the corresponding patch
file no longer exists.
---
 src/Misc/Bank.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Misc/Bank.cpp b/src/Misc/Bank.cpp
index 9a08709d..cdd21149 100644
--- a/src/Misc/Bank.cpp
+++ b/src/Misc/Bank.cpp
@@ -200,6 +200,9 @@ int Bank::savetoslot(unsigned int ninstrument, Part *part)
         return err;
     addtobank(ninstrument, legalizeFilename(tmpfilename) + ".xiz",
               (char *) part->Pname);
+    //Since we've changed the contents of one of the banks, rescan the
+    //database to keep it updated.
+    db->scanBanks();
     return 0;
 }
 
-- 
2.47.0

